### PR TITLE
Raman test

### PR DESCRIPTION
**Pull request by aleFerrari**
This pull request introduces tests for the RamanFiber.

### CODE REVIEW

File: tests/test_science_utils.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        eqpt_params = json.load(file)
+    spectral_info_params = eqpt_params['SI'][0]
+    spectral_info_params.pop('power_dbm')
+    spectral_info_params.pop('power_range_db')
+    spectral_info_params.pop('tx_osnr')
+    spectral_info_params.pop('sys_margins')
+    spectral_info_input = create_input_spectral_information(power=power, **spectral_info_params)
+    
+    # RamanFiber 
+    with open('tests/data/raman_fiber_config.json', 'r') as file:
```
Comment (jktjkt) [inline tests/test_science_utils.py]: same here

File: tests/test_science_utils.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+from numpy.testing import assert_allclose
+from gnpy.core.info import create_input_spectral_information
+from gnpy.core.elements import RamanFiber
+from gnpy.core.network import load_sim_params
+
+#@pytest.mark.parametrize()
+def test_raman_fiber():
+    # spectral information generation
+    power = 1e-3
+    with open('tests/data/eqpt_config.json', 'r') as file:
```
Comment (jktjkt) [inline tests/test_science_utils.py]: other tests appear to support relative paths better, i.e. like this:

```py
from pathlib import Path
TEST_DIR = Path(__file__).parent

with open(TEST_DIR / 'tests' / 'data' / 'eqpt_config.json'):
   ...
```

The biggest advantage is the `__file__` statement which make this work even when running from inside another directory.

File: tests/test_science_utils.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    spectral_info_params.pop('power_dbm')
+    spectral_info_params.pop('power_range_db')
+    spectral_info_params.pop('tx_osnr')
+    spectral_info_params.pop('sys_margins')
+    spectral_info_input = create_input_spectral_information(power=power, **spectral_info_params)
+    
+    # RamanFiber 
+    with open('tests/data/raman_fiber_config.json', 'r') as file:
+                raman_fiber_params = json.load(file)
+    sim_params = load_sim_params('tests/data/sim_params.json')
```
Comment (jktjkt) [inline tests/test_science_utils.py]: ...and here as well.

**Comment by jktjkt** [comment]
I forgot to say -- I woud prefer if this PR contained just a single patch. I think that it's adding a reasonably self-contained change, so splitting it into several patches is not needed. I can squash during merge of course, but it's something to keep in mind, especially for bigger PRs where I care about individual commits.

**Comment by aleFerrari** [comment]
> I like this. I think that it works on a correct level of abstraction. I hope that you checked these results and that they look sane :).
> 
Yes, I did.

File: tests/test_science_utils.py (hunk 2 above)
Comment (aleFerrari) [reply tests/test_science_utils.py]: The commit 03168ba should fix it.

File: tests/test_science_utils.py (hunk 1 above)
Comment (aleFerrari) [reply tests/test_science_utils.py]: The commit 03168ba should fix it.

File: tests/test_science_utils.py (hunk 3 above)
Comment (aleFerrari) [reply tests/test_science_utils.py]: The commit 03168ba should fix it.

**Comment by EstherLerouzic** [comment]
Thanks a lot for the code !
I have read it and I also like it. The tests are performed on only one config of sim_params and raman_fiber_config: do you think it would make sense  to have the same test run for some other configs with a 'parametrize' (for example not the same computed channels, or grid, or type of fibers...) in order to cover other physical cases?
When I read the PR last friday, travis tests were OK, but they are no more OK today. It  seems that one test with sphinx_build is not passing today due to 
"
TypeError: '_traversal_list' object is not reversible
"

... I ran it on my computer but it says "build succeeded, 12 warnings."

**Comment by jktjkt** [comment]
> I have read it and I also like it. The tests are performed on only one config of sim_params and raman_fiber_config: do you think it would make sense to have the same test run for some other configs with a 'parametrize' (for example not the same computed channels, or grid, or type of fibers...) in order to cover other physical cases?

I agree that parametrization would be nice. However, I vote for taking this code as-is because it's an improvement already. And I of course encourage @aleFerrari to submit follow-up commits or PRs which make this parametrized.

> When I read the PR last friday, travis tests were OK, but they are no more OK today.

Fixed that via #323.

**Comment by jktjkt** [comment]
>  Codacy/PR Quality Review — [Not up to standards. This pull request quality could be better.](
https://app.codacy.com/app/jktjkt/oopt-gnpy/pullRequest?prid=4613475)

At least some of the reported issues look legit, can you please take a look as well, @aleFerrari ?

**Comment by aleFerrari** [comment]
> > I have read it and I also like it. The tests are performed on only one config of sim_params and raman_fiber_config: do you think it would make sense to have the same test run for some other configs with a 'parametrize' (for example not the same computed channels, or grid, or type of fibers...) in order to cover other physical cases?
> 
> I agree that parametrization would be nice. However, I vote for taking this code as-is because it's an improvement already. And I of course encourage @aleFerrari to submit follow-up commits or PRs which make this parametrized.

I agree with Jan, anyway I'm going to add new tests for the RamanFiber.

**Comment by aleFerrari** [comment]
> > Codacy/PR Quality Review — [Not up to standards. This pull request quality could be better.](https://app.codacy.com/app/jktjkt/oopt-gnpy/pullRequest?prid=4613475)
> 
> At least some of the reported issues look legit, can you please take a look as well, @aleFerrari ?
I did with commits 2ef2af3, 06cbcce and fc307f3.

File: tests/test_science_utils.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+from gnpy.core.network import load_sim_params
+from pathlib import Path
+TEST_DIR = Path(__file__).parent
+
+def test_raman_fiber():
+    """ Test the accuracy of propagating the RamanFiber.
+    """
+    # spectral information generation
+    power = 1e-3
+    print(TEST_DIR)
```
Comment (jktjkt) [inline tests/test_science_utils.py]: nit: this line can be removed

**Pull request merged**